### ISSUE TITLE
Let It crash option

### ISSUE THREAD

**Issue by HurricanKai** [aspnet/AspNetCore#11249]
### Is your feature request related to a problem? Please describe.
I'm deploying a Blazor Project to Kubernetes, and noticed that there is no way to let an instance fail if an error occures, this would be benefical, because then Kubernetes (or another tool) can handle the failure, instead of the instance staying in an unusable state.
In my case the underlying Redis connection was closed due to the Redis node failing, which put the ASP.Net Core instance into an unusable state, unable to serve the website, but also not failing. This applies to any ASP.Net Core based Project, not only Blazor Projects.

### Describe the solution you'd like
Some kind of configurable bool that let's the Application Exit when Exceptions are cought.
Alternatively there could be some kind of Health endpoint, but I'd imagine that being out of scope for ASP.Net Core.

**Comment by anurse** [comment]
Redis failures can be transient so SignalR intentionally doesn't fail the application when a connection error occurs. It is currently possible to detect when Redis fails and perform your own logic (terminating the application or otherwise signalling to other components that the connection is non-functional). We have some docs on how to do this here: https://docs.microsoft.com/en-us/aspnet/core/signalr/redis-backplane?view=aspnetcore-2.2#custom-behavior-for-connection-failures . If that workaround doesn't work for your scenario, please let us know!

Basically you can provide your own Redis connection and hook the `ConnectionFailed` event. In there, you could do whatever logic you want.

It seems useful to provide a first-class way to do this though so I filed https://github.com/aspnet/AspNetCore/issues/11250 to help track that.

Putting this in the Blazor area now to see if there's anything further they want to do with it in the short-term.

**Comment by HurricanKai** [comment]
I did actually not know that SignalR can manage Redis, I was using it separatly, injecting

**Comment by anurse** [comment]
So you have your own Redis components in place? It's up to you how you handle errors in that. If you want to stop the application when Redis goes down, you can do so already with [`IApplicationLifetime.StopApplication`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.hosting.iapplicationlifetime?view=aspnetcore-2.2).

**Comment by HurricanKai** [comment]
I see, I'll try implement it using that. Thanks for the help 😄

**Comment by anurse** [comment]
@HurricanKai that's probably your best option. ASP.NET Core already puts an `IApplicationLifetime` implementation in DI, so you just need to inject it into the constructor of your service/controller/etc. and you can call `StopApplication` on it to stop the app.

**Comment by rynowak** [comment]
A better option would probably be to use health checks https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks?view=aspnetcore-2.2
